tests/integration/conftest.py

```diff
@@ -0,0 +1,243 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Default fixtures for functional tests."""
+
+from functools import partial
+from pathlib import Path
+import re
+from time import sleep
+from shutil import rmtree
+
+import pytest
+
+from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
+from cylc.flow.wallclock import get_current_time_string
+
+from . import (
+    _expanduser,
+    _rm_if_empty,
+    _make_flow,
+    _make_scheduler,
+    _run_flow
+)
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    """Expose the result of tests to their fixtures.
+
+    This will add a variable to the "node" object which differs depending on
+    the scope of the test.
+
+    scope=function
+        `_function_outcome` will be set to the result of the test function.
+    scope=module
+        `_module_outcome will be set to a list of all test results in
+        the module.
+
+    https://github.com/pytest-dev/pytest/issues/230#issuecomment-402580536
+
+    """
+    outcome = yield
+    rep = outcome.get_result()
+
+    # scope==function
+    item._function_outcome = rep
+
+    # scope==module
+    _module_outcomes = getattr(item.module, '_module_outcomes', {})
+    _module_outcomes[(item.nodeid, rep.when)] = rep
+    item.module._module_outcomes = _module_outcomes
+
+
+def _pytest_passed(request):
+    """Returns True if the test(s) a fixture was used in passed."""
+    if hasattr(request.node, '_function_outcome'):
+        return request.node._function_outcome in {'passed', 'skipped'}
+    return all((
+        report.outcome in {'passed', 'skipped'}
+        for report in request.node.obj._module_outcomes.values()
+    ))
+
+
+def _rmtree(path, tries=5):
+    """Remove a directory and its contents.
+
+    Attempt to remove a directory up to ``tries`` times before failing.
+
+    This should be used in-place of ``shutil.rmtree`` which doesn't
+    work reliably on NFS.
+
+    See also:
+        * https://stackoverflow.com/questions/58943374/
+          shutil-rmtree-error-when-trying-to-remove-nfs-mounted-directory
+        * https://bugzilla.redhat.com/show_bug.cgi?id=1362667
+
+    Args:
+        path (pathlib.Path): The directory to remove.
+        tries (int): Maximum number of attempts to make.
+
+    """
+    exc = None
+    for _ in range(tries):
+        try:
+            rmtree(path)
+            return
+        except OSError as exc:
+            # give NFS some thinking time
+            sleep(0.01)
+    raise exc
```

Comment:
This fixes a bug spotted by @datamel where `shutul.rmtree` fails on NFS.

It's a nasty hack but it does the job, here are some examples of how other projects have handled this issue:

* https://github.com/hashdist/hashdist/pull/116/files#diff-f8e43180100ba3086d46440b72e3e8f3
* https://github.com/davidemms/OrthoFinder/commit/3d56a6e4870058f78721fa733ae27f60ab4d0c21#diff-4b6271e9a0c35f1d832125bcadf25034